weave-js/src/components/PagePanelComponents/Home/Browse3/pages/MonitorsPage/CreateMonitorDrawer.tsx

```diff
@@ -0,0 +1,488 @@
+  const {useCalls} = useWFHooks();
+  const [scorers, setScorers] = useState<{name: string; refUri?: string}[]>([]);
+  const [active, setActive] = useState<boolean>(false);
+  const [isCreating, setIsCreating] = useState<boolean>(false);
+  const {sortBy, lowLevelFilter} = useFilterSortby(filter, {items: []}, [
+    {field: 'started_at', sort: 'desc'},
+  ]);
+  useMemo(() => {
+    const existingScorers = (monitor?.val['scorers'] || []).map(
+      (scorerRefUri: string) => ({
+        refUri: scorerRefUri,
+        name: parseRef(scorerRefUri).artifactName,
+      })
+    );
+    setMonitorName(monitor?.val['name'] || '');
+    setDescription(monitor?.val['description'] || '');
+    setSamplingRate((monitor?.val['sampling_rate'] || 0.1) * 100);
+    setFilter({opVersionRefs: monitor?.val['op_names'] || []});
+    setScorers(existingScorers);
+    setActive(monitor?.val['active'] || false);
+    setFilterModel(
+      queryToGridFilterModel(monitor?.val['query']) || {
+        items: [],
+      }
+    );
+  }, [monitor, setMonitorName]);
+
+  const {result: callsResults, loading: callsLoading} = useCalls(
+    entity,
+    project,
+    lowLevelFilter,
+    PAGE_SIZE,
+    PAGE_OFFSET,
+    sortBy
+  );
+
+  const tableData = useMemo(() => {
+    if (callsLoading || callsResults === null) {
+      return [];
+    }
+    return prepareFlattenedCallDataForTable(callsResults);
+  }, [callsResults, callsLoading]);
+
+  const {columns} = useCallsTableColumns(
+    entity,
+    project,
+    filter,
+    tableData,
+    new Set(),
+    () => {},
+    () => {},
+    () => false
+  );
+
+  const handleNameChange = useCallback(
+    (value: string) => {
+      setMonitorName(value);
+      const validationResult = validateDatasetName(value);
+      setNameError(validationResult.error);
+    },
+    [setNameError]
+  );
+
+  const opVersionOptions = useOpVersionOptions(entity, project, {});
+
+  useMemo(() => {
+    setSelectedOpVersionOption(filter.opVersionRefs ?? []);
+  }, [filter]);
+
+  const enableCreateButton = useMemo(() => {
+    return (
+      monitorName.length > 0 &&
+      selectedOpVersionOption.length > 0 &&
+      scorers.length > 0 &&
+      nameError === null
+    );
+  }, [monitorName, selectedOpVersionOption, scorers, nameError]);
+
+  const objCreate = useObjCreate();
+
+  const createScorer = useCallback(
+    async ({
+      name: scorerName,
+      refUri,
+    }: {
+      name: string;
+      refUri?: string;
+    }): Promise<WeaveObjectVersionKey> => {
+      let scorerDigest: string;
+      if (refUri) {
+        // If the scorer already exists (this is an edit) we do not recreate it.
+        scorerDigest = parseRef(refUri).artifactVersion;
+      } else {
+        // If refUri is undefined the scorer needs to be create.
+        const scorerClassName = scorerName;
+        scorerName = `${scorerName}-${uuidv4().split('-')[0]}`;
+
+        const scorerObj = {
```

Comment:
this structure is implicitly bound to the payload that the python client creates. It would be nice for that to be extracted into an specific module so that it is not buried inside component implementation. For example: `ref: None` is a BUG in python! Fixing this should be centralized.

Reply:
I'll move this out to `tsDataModelHooks.ts` in a follow-up PR.

Re: `ref: None`, I tried publishing an `Evaluation` and a `Scorer` from the Python SDK and they also had `ref: None` in the `val` field. 🤔